src/appengine/libs/issue_management/github.py

```diff
@@ -0,0 +1,177 @@
+
+ISSUE_TITTLE_TEXT = "OSS-Fuzz issue {bug_information}"
+
+ISSUE_CONTENT_TEXT = ("OSS-Fuzz has found a bug in this project. Please see "
+                      f"{TESTCASE_REPORT_URL}"
+                      "for details and reproducers."
+                      "\n\n"
+                      "This issue is mirrored from "
+                      f"{MONORAIL_URL} "
+                      "and will auto-close if the status changes there."
+                      "\n\n"
+                      "If you have trouble accessing this report, "
+                      f"please file an issue at {OSS_FUZZ_ISSUE_URL}."
+                      "\n")
+
+ISSUE_ClOSE_COMMENT_TEXT = ("OSS-Fuzz has closed this bug. Please see "
+                            f"{MONORAIL_URL} "
+                            "for details.")
+
+
+def get_issue_title(testcase):
+  """Generate the title of the issue"""
+  return ISSUE_TITTLE_TEXT.format(bug_information=testcase.bug_information)
+
+
+def get_issue_body(testcase):
+  """Generate the body of the issue"""
+  return ISSUE_CONTENT_TEXT.format(
+      domain=data_handler.get_domain(),
+      testcase_id=testcase.key.id,
+      bug_information=testcase.bug_information)
+
+
+def get_issue_close_comment(testcase):
+  """Generate the closing comment of the issue"""
+  return ISSUE_ClOSE_COMMENT_TEXT.format(
+      bug_information=testcase.bug_information)
+
+
+def get_access():
+  """Get access to GitHub with the oss-fuzz personal access token"""
+  token = db_config.get_value("oss_fuzz_robot_github_personal_access_token")
+  if not token:
+    logs.log_error("Unable to get oss-fuzz-robot's personal access token.")
+    return None
+  return github.Github(token)
+
+
+def filing_enabled(testcase):
+  """Check if the project YAML file requires to file a GitHub issue."""
+  require_issue = data_handler.get_value_from_job_definition(
+      testcase.job_type, 'FILE_GITHUB_ISSUE', default='False')
+  return require_issue.lower() == 'true'
+
+
+def get_repo(testcase, access):
+  """Get the GitHub repository to file the issue"""
+  repo_url = data_handler.get_value_from_job_definition(testcase.job_type,
+                                                        'MAIN_REPO', '')
+  if not repo_url:
+    logs.log_error("Unable to fetch the MAIN_REPO URL from job definition.")
+    return None
+  repo_name = repo_url.removeprefix('https://github.com/')
+
+  try:
+    target_repo = access.get_repo(repo_name)
+  except github.UnknownObjectException:
+    logs.log_error(f"Unable to locate GitHub repository "
+                   f"named {repo_name} from URL: {repo_url}.")
+    target_repo = None
+  return target_repo
+
+
+def post_issue(repo, testcase):
+  """Post the issue to the Github repo of the project."""
+  issue_title = get_issue_title(testcase)
+  issue_body = get_issue_body(testcase)
+  return repo.create_issue(title=issue_title, body=issue_body)
+
+
+def update_testcase_properties(testcase, repo, issue):
```

Comment:
This can be private too. Prefix the function name with "_". Can evaluate all other functions in this module to determine the minimal ones that need to be public.

Reply:
Fixed with [69b1598](https://github.com/google/clusterfuzz/pull/2578/commits/69b159841bc32aefc24d178eb9cc81577ece02c7).